### ISSUE TITLE
Function does not accept argument {i = _}

### ISSUE THREAD

**Issue by MirceaS** [agda/agda-categories#60]
Hi!
I just cloned and tried to load your library but I get the error:

.../agda-categories/Categories/Morphism/Isomorphism.agda:181,41-51
Function does not accept argument {i = _}
when checking that {i = _ ∼⁺⟨ [ f ] ⟩ [ g ]} {j = [ h ]} eq are
valid arguments to a function of type
{A B : Obj} → ≅⁺⇒⇒⁺ Preserves _≃⁺_ ⟶ _≈⁺_

Any idea why that might be and what I can do to make the library loadable?
Thank you!

**Comment by JacquesCarette** [comment]
This seems to somehow be related to the version of Agda you're using -- 2.6.0.1 ?  I'll look at fixing it for all versions, but figuring out which version would be a good start.

**Comment by MirceaS** [comment]
> This seems to somehow be related to the version of Agda you're using -- 2.6.0.1 ? I'll look at fixing it for all versions, but figuring out which version would be a good start.

It's 2.6.0.1, yeah. I didn't test it on any other version.

**Comment by JacquesCarette** [comment]
I think I know what's going on - I'm testing a fix.

**Comment by MirceaS** [comment]
I thought this library was specifically redesigned for 2.6. Is 2.6.0.1 not the latest version of agda?
Thanks for the quick feedback, by the way!

**Comment by JacquesCarette** [comment]
It is supposed to work on 2.6 and 2.6.0.1, definitely. It seems that there's a weird version-specific issue about naming of implicits. It's probably a weird bug in Agda, but I have what I think is a workaround that I'm testing now.

**Comment by JacquesCarette** [comment]
If you pull the master branch now, it should be fixed.

**Comment by MirceaS** [comment]
It's working now! Thanks for the fix!

**Comment by MirceaS** [comment]
I noticed that you put in some empty implicit arguments. Does that mean that giving specific, named implicit arguments is not supported?

**Comment by JacquesCarette** [comment]
I first fixed it that way because 'it works', and then I went back to change one of the definitions to make the implicits have a fixed name (or so I think I've managed that). I think the deeper problem is that using `variable` declarations seems to generate declarations that vary in order. [Your report is not the first one with this problem.]  Unfortunately, I can't reproduce it on any of my machines, so it is quite difficult to diagnose.

What would help is if you showed me the type of `≅*⇒⇒*-cong` that Agda reports (make the `begin `... a hole and ask there).  I am guessing that either
1. `i` is not named `i` but something else,
2. `i` occurs in a different spot than 3rd.
Figuring that out might help generate a reproducible case, that could generate an agda bug report.
